### PR TITLE
Make private key optional in TlsConnectorConfig

### PR DESCRIPTION
**Pull request by conorbros**

### CODE REVIEW

**Comment by XA21X** [comment]
The private key corresponds to the certificate, so `certificate_path` should also be optional?

File: shotover-proxy/example-configs/redis-cluster-tls/docker-compose-with-key.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      - 'REDIS_CLUSTER_REPLICAS=1'
+      - 'REDIS_CLUSTER_CREATOR=yes'
+      - 'ALLOW_EMPTY_PASSWORD=yes'
+      - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2 redis-node-3 redis-node-4 redis-node-5'
+      - 'REDIS_TLS_PORT=6379'
+      - 'REDIS_TLS_ENABLED=true'
+      - 'REDIS_TLS_CERT_FILE=/usr/local/etc/redis/certs/redis.crt'
+      - 'REDIS_TLS_KEY_FILE=/usr/local/etc/redis/certs/redis.key'
+      - 'REDIS_TLS_CA_FILE=/usr/local/etc/redis/certs/ca.crt'
+      - 'REDIS_TLS_AUTH_CLIENTS=false'
```
Comment (rukai) [inline shotover-proxy/example-configs/redis-cluster-tls/docker-compose-with-key.yml:87]: Why is this here?

Comment (conorbros) [reply shotover-proxy/example-configs/redis-cluster-tls/docker-compose-with-key.yml:87]: Are you talking about `- 'REDIS_TLS_AUTH_CLIENTS=false'`?

Comment (rukai) [reply shotover-proxy/example-configs/redis-cluster-tls/docker-compose-with-key.yml:87]: yeah

**Review by rukai — APPROVED** [review]
Nice!
LGTM

**Review by benbromhead — APPROVED** [review]

**Comment by benbromhead** [comment]
Fixes https://github.com/shotover/shotover-proxy/issues/708
